src/modules/data/sound.c

```diff
@@ -25,13 +25,15 @@ struct Sound {
   uint32_t sampleRate;
   uint32_t frames;
   uint32_t cursor;
+  void *callbackMemo;
+  SoundDestroyCallback callbackMemoDestroy;
```

Comment:
For informational purposes can you explain what the memo/destroy callback are being used for?  At some point it may help to know how it's being used.

Reply:
You mean here or in a comment?

callbackMemo is for any arbitrary "state" associated with the read callback. For example imagine a Sound which is a sine wave generator. callbackMemo could be a pointer to a float representing the sine frequency.

callbackMemoDestroy is to free the memory used by callbackMemo.

In my current usage, I have a Sound whose callback refers to static [global] state for my sound effects. So callbackMemo is NULL, but callbackMemoDestroy is used (it sets a flag in the static structure noting that sound playback has stopped). Tomorrow I'm going to do PureData hookup and callbackData will point to a structure for the PureData playback channel associated with a particular Sound source (a buffer and a pointer to the generator, since one generator will generate several channels at once once per audio frame) but the callbackMemoDestroy might be blank because in that case the playback channels have their memory managed elsewhere.

I'll add a comment.